plugins/out_opentelemetry/opentelemetry.c

```diff
@@ -1027,30 +1027,30 @@ static int process_traces(struct flb_event_chunk *event_chunk,
     flb_plg_debug(ctx->ins, "ctraces msgpack size: %lu",
                   event_chunk->size);
 
-    ret = ctr_decode_msgpack_create(&ctr,
-                                    (char *) event_chunk->data,
-                                    event_chunk->size, &off);
-    if  (ret != ok) {
-        flb_plg_error(ctx->ins, "Error decoding msgpack encoded context");
-        result = FLB_ERROR;
-        goto exit;
-    }
-
-    /* Create a OpenTelemetry payload */
-    encoded_chunk = ctr_encode_opentelemetry_create(ctr);
-    if (encoded_chunk == NULL) {
-        flb_plg_error(ctx->ins,
-                      "Error encoding context as opentelemetry");
-        result = FLB_ERROR;
-        goto exit;
-    }
+    while (ctr_decode_msgpack_create(&ctr,
+                                     (char *) event_chunk->data,
+                                     event_chunk->size, &off) == ok) {
+        /* Create a OpenTelemetry payload */
+        encoded_chunk = ctr_encode_opentelemetry_create(ctr);
+        if (encoded_chunk == NULL) {
+            flb_plg_error(ctx->ins,
+                          "Error encoding context as opentelemetry");
+            result = FLB_ERROR;
+            goto exit;
+        }
 
-    /* concat buffer */
-    flb_sds_cat_safe(&buf, encoded_chunk, flb_sds_len(encoded_chunk));
+        /* concat buffer */
+        ret = flb_sds_cat_safe(&buf, encoded_chunk, flb_sds_len(encoded_chunk));
+        if (ret != 0) {
+            flb_plg_error(ctx->ins, "Error appending encoded trace to buffer");
```

Comment:
Release the trace context here, even though `flb_sds_cat_safe` failed `ctr_encode_opentelemetry_create` succeeded. Otherwise this will leak.